depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp

```diff
@@ -43,15 +46,75 @@ class Guard {
  * @brief Reads a packet from an XLinkStream and frees it on destruction
  * @note Make sure to call `getData` or else the packet will go un-processed
  */
-struct PacketReader {
+class PacketReader {
+public:
     PacketReader(dai::XLinkStream& stream)
             : _stream(stream) {
         _packet = _stream.readRaw();  // blocking read
+
+        _msgpack_size = static_cast<uint32_t>(fromLE(_packet->data + _packet->length - 4));
+        if (_msgpack_size < 0) {
+            throw std::runtime_error("Bad packet, couldn't parse");
+        }
```

Comment:
Isn't that impossible? msgpack_size by definition of type can't have negative values. Maybe compare against the size of data read.
Eg: total_size >= msgpack.size